designs/credentialManagement/credentialManagement.md

```diff
@@ -0,0 +1,139 @@
+# Credential Management
+
+## Abstract
+
+This document describes the process of managing and switching of AWS credentials within the toolkit.
+
+## Motivation
+
+AWS provides many different ways to retrieve the credentials required to make calls to AWS including but not limited to:
+credential/config files, environment variables, system properties, HTTP metadata APIs. A user may also assume roles using 
+a source set of credentials or use MFA. Due to this large matrix of possibilities the toolkit strives to provide an 
+intuitive abstraction layer on top of some of these sources so that the implementation details are hidden and a 
+consistent user experience can be provided, while also enabling future expansion of new sources, including possibly by 
+third party plugins.
+
+## Classes and Concepts
+![ClassDiagram]
+
+1. `AwsRegion` - Value class that represents an AWS Region and joins together related data for that region. This data is sourced from the endpoints.json file.
+It contains the following data:
+    1. `ID` - Contains the ID of the region  (e.g. `us-west-2`)
+    1. `Name` - Contains the human readable name for the region (e.g. `US West (Oregon)`)
+    1. `Partiton ID` - Contains the ID of the top level AWS partition (e.g. `aws`, `aws-cn`)
+
+1. `CredentialIdentifier` - Represents the globally unique identifier for a possible credential profile in the toolkit. This identifier must be deterministic 
+and represent the same underlying source across IDE sessions. This is shown to the user as the **Profile** in the UI.
+
+1. `AwsCredentialsProvider` - SDK interface that resolves AWS Credentials from the provider. For more info, see [AwsCredentialsProvider] in the SDK.
```

Comment:
Not sure if we want to split stuff like this into its own section. Like a "this implementation" vs "generic ideal credentials management"